src/main/java/com/firebolt/jdbc/connection/settings/FireboltProperties.java

```diff
@@ -219,13 +222,19 @@ private static <T> T getSetting(Properties info, FireboltSessionProperty param)
 	}
 
 	public static FireboltProperties copy(FireboltProperties properties) {
-		return properties.toBuilder().additionalProperties(new HashMap<>(properties.getAdditionalProperties())).build();
+		return properties.toBuilder().runtimeAdditionalProperties(new HashMap<>(properties.getRuntimeAdditionalProperties())).build();
 	}
```

Comment:
First, thank you for comment. However I think that this is OK. Indeed, the "initial properties" are read only. They are initialized only once and therefore `properties.toBuilder().build()` creates exact copy of `properties`. 

Why this method was implemented? I is used only once in method named `changeProperty()`. This method indeed changes property and tries to run `select 1` in order to validate that the the change does not break session. If it breaks session I used wanted not to apply the new property. I have implemented this using temporary property - the copy of the initial instance with the change. But class `FireboltProperties` does not have copy constructor, so I used combination of `toBuilder().build()`. 

But there is a catch here. The copy is not deep but shallow. This means that the map that holds additional properties is the same in the temporary  instance of `FireboltProperties` and I want to avoid this by wrapping it with `new HashMap<>()`.

Reply:
Thank you for your comment.
I see. What if `changePropery` tries to validate a change in `force_pgdate_timestampntz`? Would it run SELECT 1 only with this flag or will it also have `advanced_mode` that was set on connection? If the former then it will fail.

Reply:
Well, I have reviewed the code again. If the `changeProperty()` is called with `force_pgdate_timestampntz` (or any other property) we copyt the existing FireboltProperties using `copy()` method, then add the new propety to its runtime properties list, and then validate connection using `SELECT 1`. 

If `advanced_mode` was configured using JDBC URL it presents in list of initial properties and will be used in `FireboltProperties.getAdditionalProperties()`. If it was set at runtime it is stored in `runtimeAdditionalProperties` and  will be used in `FireboltProperties.getAdditionalProperties()`.

Moreover even if it was set during initialization and then updated N times at runtime (where N>0) the last value will be used (see `FireboltProperties.getAdditionalProperties()`).

Reply:
I understand, but imagine the following scenario:
1. A connection is established with advanced_mode=1 in the URL
2. We call `changeProperty` with `force_pgdate_timestampntz=1`
3. It copies over existing properties and tries to validate `force_pgdate_timestampntz`
4. Since for copying it's using `runtimeAdditionalProperties` not `getAdditionalProperties`, `advanced_mode` is not copied over.
5. `changeProperty` returns an error.

Reply:
No, it copies everything. The only reason to mention runtimeAdditionalProperties in copy() method is guarantee that they are copied, so thier modification does not affect the source HashMap.